lib/util/itinerary.js

```diff
@@ -125,36 +174,30 @@ export function getStepStreetName (step) {
   return step.streetName
 }
 
-export function getLegModeString (leg) {
+export function getLegModeLabel (leg) {
   switch (leg.mode) {
     case 'BICYCLE_RENT': return 'Biketown'
     case 'CAR': return leg.hailedCar ? 'Ride' : 'Drive'
     case 'GONDOLA': return 'Aerial Tram'
     case 'TRAM':
       if (leg.routeLongName.toLowerCase().indexOf('streetcar') !== -1) return 'Streetcar'
       return 'Light Rail'
+    case 'MICROMOBILITY': return 'Ride'
   }
   return toSentenceCase(leg.mode)
 }
 
-export function getModeIcon (mode, customIcons) {
-  let modeStr = mode.mode || mode
-
-  // Special handling for company-specific icons
-  if (mode.company) {
-    modeStr = `${modeStr}_${mode.company.toUpperCase()}`
-  }
-
-  // Check if there is a custom icon for this mode
-  if (customIcons && modeStr in customIcons) {
-    return customIcons[modeStr]
+export function getIcon (iconId, customIcons) {
+  // Check if there is a custom icon
+  if (customIcons && iconId in customIcons) {
+    return customIcons[iconId]
```

Comment:
Just curious: what happened to the company-specific stuff?

Reply:
There is now just one icon lookup, so an iconId can be either a company or a mode.

Reply:
Can you add a comment to this method to describe that it will return an icon for either a company or a mode?